iceoryx_posh/test/moduletests/test_roudi_memory_manager.cpp

```diff
@@ -109,6 +142,15 @@ TEST_F(RouDiMemoryManager_Test, AddMemoryProviderExceedsCapacity)
     EXPECT_THAT(expectError.get_error(), Eq(RouDiMemoryManagerError::MEMORY_PROVIDER_EXHAUSTED));
 }
 
-/// @todo Add TEST_F(RouDiMemoryManager_Test, StreamOperatorTranslationIsCorrect)
+TEST_F(RouDiMemoryManager_Test, OperatorTest)
+{
+    for(int16_t i = 0; i < nbTestCase; i++)
+    {
+        iox::log::LogStream logStream(loggerMock);
+        iox::log::LogStream logStreamResult(loggerMockResult);
+        logStream << m_testCombinationRoudiMemoryManagerError[i];
+        logStreamResult << m_testResultOperatorMethod[i];
+        EXPECT_EQ(*::testing::PrintToString(logStream).c_str(), *::testing::PrintToString(logStreamResult).c_str());
```

Comment:
this is overly complicated. Theoretically you should be able to do a `EXPECT_THAT(loggerMock.m_logs[0].message, Eq(m_testResultOperatorMethod[i]));`

Reply:
Theoretically, but I am getting a segmentation fault.

Reply:
The reason for this is that the LogStream object sends the data in its destructor to the logger. Since you don't use a temporary object, the logger receives the data only at the end of the scope, which is too late.

You have three options
- use `logStram.Flush()` to send the data to the logger
- move the following to an own scope
```
iox::log::LogStream logStream(loggerMock);
logStream << m_testCombinationRoudiMemoryManagerError[i];
```
- use a temporary object `iox::log::LogStream(loggerMock) << "" << m_testCombinationRoudiMemoryManagerError[i];` (unfortunately there is a flaw in the Logger and the first item to log with a temporary object must be a type with built in logging support like an empty string`""`)

Additionally, the logger mock keeps all messages until destruction, so you either have to move the logger mock into the for loop or have to check for `loggerMock.m_logs[i].message`. There should also be a sanity check that `loggerMock.m_logs` is large enough to index.

I hope I could give you a better understanding how the logging stuff works